BOUNTY.md

```diff
@@ -0,0 +1,116 @@
+of GraphQL.NET for compensation ("projects"). These projects are determined by GraphQL.NET maintainers to have the
+potential for high-impact improvement to the product. Design proposals that would fulfill a bounty may
+be made either by individuals or teams of developers, but the final payout will be made to only one stakeholder.
+Routine bug fixes and similar pull requests do not fall into this program at this time.
+
+
+## Bounty Process Flowchart
+
+1. The GraphQL.NET team publishes a Request For Proposals (RFP) describing a feature that the team has decided
+   should be prioritized, or a bug that is in need of being researched and fixed (a "project"). This project
+   will be posted as a GitHub issue and will indicate the amount of the bounty in US dollars (USD).
+     * The GitHub issue is labeled with **`bounty`**.
+2. Developers write proposals for how they would implement the feature or fix the bug described in the RFP.
+3. The GraphQL.NET team works with developers to clarify and refine their proposals.
+4. A proposal is accepted. The developer(s) responsible for the proposal work to implement the feature or
+   bug fix according to their proposal.
+    * The GitHub issue is also labeled with **`in progress`**.
+5. Pull requests are submitted to implement the proposal by the developer(s).
+    * The GitHub pull request is linked to the issue.
+6. Pull requests are reviewed by project maintainers, and the pull request developers make any changes to the
+   pull requests accordingly.
+7. Pull requests are merged into the `master` or `develop` branch.
+8. Developer(s) submit invoice to the [GraphQL.NET Open Collective](https://opencollective.com/graphql-net)
+   for bounty amount, confirmed by GraphQL.NET team members.
+9. Payment disbursed to developer(s) by the Open Source Collective from GraphQL.NET funds.
+    * The GitHub issue is labeled with **`bounty-paid`** and the aforementioned labels are removed.
+
+
+## How to Claim a Bounty
+
+We've created a process to ensure multiple contributors aren’t competing on the same project, and to make sure
+work is properly merged into GraphQL.NET once completed.
+
+1. Read the GitHub issue containing the project's Request For Proposals completely.
+2. If you want to move forward, submit an proposal to the GraphQL.NET team on your suggested implementation by
+   replying to the issue. Be sure to include as many details as requested by the spec, and be sure to address
+   each of the completion criteria. For proposals that include UI changes, mockups are strongly encouraged.
+3. Work with the GraphQL.NET team to finalize plans for the best implementation. Proposals will be judged on their
+   maintainability, design quality, and adherence to spec. Note that other developers may submit competing proposals
+   at this time as well.
+4. If your proposal is accepted, you may begin code implementation that adheres to your proposal.
```

Comment:
>  may submit competing proposals at this time as well.
> If your proposal is accepted

So.. can other competing proposals arise after this one was marked as accepted (in progress)?

Reply:
The deadline written in the bounty program should help a lot preventing these situations.

Reply:
"do what we want" maybe obvious for us but not so obvious for someone who try to contribute. Here my only intent is to make things as clear as possible. Your explanation is good and I would like to move these words into document (even in the form of non-normative notice). If you do not want, then okay.

Reply:
Hmm..  I don't think we should say anything here.  We can write a separate document for internal procedures that we can use as guidelines in these scenarios, if the need arises.  This separate document should only be visible to maintainers.